.idea/codeStyles/Project.xml

```diff
@@ -0,0 +1,57 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
+    <option name="RIGHT_MARGIN" value="80" />
+    <Objective-C>
+      <option name="HEADER_GUARD_STYLE_PATTERN" value="${PROJECT_REL_PATH}_${FILE_NAME}_${EXT}_" />
+      <option name="NAMESPACES_NAMING_CONVENTION">
+        <value prefix="" style="LOWERCASE" suffix="" />
+      </option>
+      <option name="CLASSES_AND_STRUCTS_NAMING_CONVENTION">
+        <value prefix="" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="ENUMS_NAMING_CONVENTION">
+        <value prefix="" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="TYPEDEFS_NAMING_CONVENTION">
+        <value prefix="" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="UNIONS_NAMING_CONVENTION">
+        <value prefix="" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="METHODS_NAMING_CONVENTION">
+        <value prefix="" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="FIELDS_NAMING_CONVENTION">
+        <value prefix="" style="SNAKE_CASE" suffix="_" />
+      </option>
+      <option name="GLOBAL_FUNCTIONS_NAMING_CONVENTION">
+        <value prefix="" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="GLOBAL_VARIABLES_NAMING_CONVENTION">
+        <value prefix="k" style="PASCAL_CASE" suffix="" />
+      </option>
+      <option name="PARAMETERS_NAMING_CONVENTION">
+        <value prefix="" style="SNAKE_CASE" suffix="" />
+      </option>
+      <option name="LOCAL_VARIABLES_NAMING_CONVENTION">
+        <value prefix="" style="SNAKE_CASE" suffix="" />
+      </option>
+    </Objective-C>
+    <Objective-C-extensions>
+      <extensions>
+        <pair source="cpp" header="h" fileNamingConvention="SNAKE_CASE" />
+        <pair source="c" header="h" fileNamingConvention="SNAKE_CASE" />
+      </extensions>
+    </Objective-C-extensions>
+    <codeStyleSettings language="CMake">
+      <option name="SPACE_BEFORE_IF_PARENTHESES" value="false" />
+      <option name="SPACE_BEFORE_WHILE_PARENTHESES" value="false" />
+      <option name="SPACE_BEFORE_FOR_PARENTHESES" value="false" />
+      <indentOptions>
+        <option name="INDENT_SIZE" value="2" />
+        <option name="CONTINUATION_INDENT_SIZE" value="2" />
+        <option name="TAB_SIZE" value="2" />
+      </indentOptions>
+    </codeStyleSettings>
+  </code_scheme>
+</component>
```

Comment:
Why are you adding the files in `.idea`.  I don't think they should be.

Reply:
The plan was to add a few specific files (that are not user-specific) so that anybody using CLion will have some good code style settings.

Reply:
I don't know if we want to start supporting specific IDE.  It will create extra work to maintain as these files change with different versions of the tools.  Also, which IDE do we want to support.  Why CLion and not another?  

We use the google style.  For me, clion seems to default to that style, or something close to it. 
 That is probably because I have a google-centric version.  You can also add clang-format as an external tool, which is what I do.  Then I can format the current file by pressing Ctrl-L (or whatever you want).

Also, we want one source of truth for automatic formatting code: clang-format.  What if we can't get the two tools to do the exact same thing.

I don't think it is a profitable for us to maintain these files.

Reply:
Yes, I wanted to see what you thought. Note that CLion does indeed defer to clang format for me. These settings just define some naming conventions, which are not covered by clang format. I also wanted to add my "new file" template, but then the copyright header would be "Google LLC" by default, which doesn't seem fair, and CLion doesn't yet support a way to use a "company name" macro in file templates. Regardless, I will remove these.